src/Service/Resolvers/CosmosMutationEngine.cs

```diff
@@ -68,6 +80,80 @@ private async Task<JObject> ExecuteAsync(IDictionary<string, object?> queryArgs,
             return response.Resource;
         }
 
+        /// <summary>
+        /// Authorization check on mutation fields provided in a GraphQL Mutation request.
+        /// </summary>
+        /// <param name="context"></param>
+        /// <param name="parameters"></param>
+        /// <param name="entityName"></param>
+        /// <param name="mutationOperation"></param>
+        /// <exception cref="DataApiBuilderException"></exception>
+        public void AuthorizeMutationFields(
+            IMiddlewareContext context,
+            IDictionary<string, object?> parameters,
+            string entityName,
+            Config.Operation mutationOperation)
+        {
+            string role = string.Empty;
+            if (context.ContextData.TryGetValue(key: AuthorizationResolver.CLIENT_ROLE_HEADER, out object? value) && value is StringValues stringVals)
+            {
+                role = stringVals.ToString();
+            }
+
+            if (string.IsNullOrEmpty(role))
+            {
+                throw new DataApiBuilderException(
+                    message: "No ClientRoleHeader available to perform authorization.",
+                    statusCode: HttpStatusCode.Unauthorized,
+                    subStatusCode: DataApiBuilderException.SubStatusCodes.AuthorizationCheckFailed);
+            }
+
+            List<string> inputArgumentKeys;
+            if (mutationOperation != Config.Operation.Delete)
+            {
+                inputArgumentKeys = BaseSqlQueryStructure.GetSubArgumentNamesFromGQLMutArguments(MutationBuilder.INPUT_ARGUMENT_NAME, parameters);
+            }
+            else
+            {
+                inputArgumentKeys = parameters.Keys.ToList();
+            }
+
+            bool isAuthorized; // False by default.
+
+            switch (mutationOperation)
+            {
+                case Config.Operation.UpdateGraphQL:
+                    isAuthorized = _authorizationResolver.AreColumnsAllowedForOperation(entityName, roleName: role, operation: Config.Operation.Update, inputArgumentKeys);
+                    break;
+                case Config.Operation.Create:
+                    isAuthorized = _authorizationResolver.AreColumnsAllowedForOperation(entityName, roleName: role, operation: mutationOperation, inputArgumentKeys);
+                    break;
+                case Config.Operation.Execute:
+                case Config.Operation.Delete:
+                    // Authorization is not performed for the 'execute' operation because stored procedure
```

Comment:
This seems to be remnant of copying from Sql. Can you please remove `Operation.Execute` altogether, since I dont think for CosmosDb_nosql that operation makes sense.

Reply:
Thank you for pointing this out, removed it.